### PR TITLE
feat: account renaming via rpc

### PR DESCRIPTION
**Pull request by meeh0w**
## Description
* [CP-8963](https://ava-labs.atlassian.net/browse/CP-8963)
* Related PR: https://github.com/ava-labs/extension-avalanche-playground/pull/50

## Changes
* Adds `avalanche_renameAccount` request handler
* Migrates UI to use the new handler instead of `ExtensionRequest.ACCOUNT_RENAME` and removes the old handler
* **Maintenance:** _replaces `isCoreWeb()` with `isSyncDomain()`, as the latter also supports feature branch deploys of Core Web (e.g. `develop.core-web.pages.dev)`_

## Testing
* Use playground to rename the account (see PR link above)
* Rename account in the extension itself

## Screenshots:
_N/A_

## Checklist for the author
- [x] I've covered new/modified business logic with Jest test cases.
- [x] I've tested the changes myself before sending it to code review and QA.

### CODE REVIEW

File: src/background/services/accounts/handlers/avalanche_renameAccount.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    const account = this.accountsService.getAccountByID(accountId);
+
+    if (!account) {
+      return {
+        ...request,
+        error: ethErrors.rpc.invalidParams({
+          message: 'No account with specified ID',
+        }),
+      };
+    }
```
Comment (bferenc) [inline src/background/services/accounts/handlers/avalanche_renameAccount.ts:54]: Can we also validate the `newName` param? Currently, the handler accepts empty string.
Would be great if we added the same validation to our own UI as well (it also accepts empty strings)

File: src/background/services/accounts/handlers/avalanche_renameAccount.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+    if (isSyncDomain(request.site.domain)) {
```
Comment (bferenc) [inline src/background/services/accounts/handlers/avalanche_renameAccount.ts:56]: Should we add a check to `isSyncDomain` so it returns `true` only if the request is coming from the `active` tab?
This way synced domains won't be able to make unnoticed changes in the users' wallets.

**Review by bferenc — CHANGES_REQUESTED** [review]
Nice! Some questions below 👇

File: src/background/services/accounts/handlers/avalanche_renameAccount.ts (hunk 1 above)
Comment (meeh0w) [reply src/background/services/accounts/handlers/avalanche_renameAccount.ts:63]: Sure thing :) 6a2d3dded

File: src/background/services/accounts/handlers/avalanche_renameAccount.ts (hunk 2 above)
Comment (meeh0w) [reply src/background/services/accounts/handlers/avalanche_renameAccount.ts]: Good point! 0ab8239b6

**Review by bferenc — DISMISSED** [review]
💯

**Comment by meeh0w** [comment]
@bferenc I had to resolve conflicts 🙏

**Review by bferenc — APPROVED** [review]

**Comment by vvava** [comment]
Don't we want to accept account indexes as well? I mean not just account ids.

**Review by vvava — APPROVED** [review]

**Pull request merged**